From f8a7e0f9e1537c90ec3dfa0bb09b8b6ed669ede5 Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Sun, 22 Jun 2025 04:18:31 +0200
Subject: [PATCH 01/25] Fix libc/sys/config.h

Signed-off-by: Salar Samani <soltanisamani@gmail.com>
---
 newlib/libc/include/sys/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/include/sys/config.h b/newlib/libc/include/sys/config.h
index 5dcc77a..ee08d1a 100644
--- a/newlib/libc/include/sys/config.h
+++ b/newlib/libc/include/sys/config.h
@@ -256,7 +256,7 @@
    for read()/write() is "ssize_t" but legacy newlib code has been using
    "int" for some time.  If not specified, "int" is defaulted.  */
 #ifndef _READ_WRITE_RETURN_TYPE
-#define _READ_WRITE_RETURN_TYPE int
+#define _READ_WRITE_RETURN_TYPE _ssize_t
 #endif
 /* Define `count' parameter of read/write routines.  In POSIX, the `count'
    parameter is "size_t" but legacy newlib code has been using "int" for some
-- 
2.43.0

