From bc8a41dc99533f47b36095304d718d310f05be9a Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Sun, 22 Jun 2025 07:14:04 +0200
Subject: [PATCH 08/29] Fix libc/include/sys/_types.h

Signed-off-by: Salar Samani <soltanisamani@gmail.com>
---
 newlib/libc/include/sys/_types.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/newlib/libc/include/sys/_types.h b/newlib/libc/include/sys/_types.h
index d1112d5..fae30a3 100644
--- a/newlib/libc/include/sys/_types.h
+++ b/newlib/libc/include/sys/_types.h
@@ -203,10 +203,12 @@ typedef	_TIMER_T_	__timer_t;
 
 #ifndef __machine_sa_family_t_defined
 typedef	__uint8_t	__sa_family_t;
+#define __machine_sa_family_t_defined
 #endif
 
 #ifndef __machine_socklen_t_defined
 typedef	__uint32_t	__socklen_t;
+#define __machine_socklen_t_defined
 #endif
 
 typedef	int		__nl_item;
-- 
2.43.0

