From 89b29e617179b1093e99ca9e1c40665873ce380f Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Sun, 22 Jun 2025 07:16:18 +0200
Subject: [PATCH 09/25] Fix libc/include/sys/_pthreadtypes.h

Signed-off-by: Salar Samani <soltanisamani@gmail.com>
---
 newlib/libc/include/sys/_pthreadtypes.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/newlib/libc/include/sys/_pthreadtypes.h b/newlib/libc/include/sys/_pthreadtypes.h
index 75e9e1c..cd7bf60 100644
--- a/newlib/libc/include/sys/_pthreadtypes.h
+++ b/newlib/libc/include/sys/_pthreadtypes.h
@@ -21,6 +21,7 @@
 #if defined(_POSIX_THREADS) || __POSIX_VISIBLE >= 199506
 
 #include <sys/sched.h>
+#include <time.h>
 
 /*
  *  2.5 Primitive System Data Types,  P1003.1c/D10, p. 19.
-- 
2.43.0

