From ba59499d9af81178f2f7ac5c933981bcada526d9 Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Sun, 29 Jun 2025 21:02:07 +0200
Subject: [PATCH 14/25] Fix libc/include/sys/param.h

---
 newlib/libc/include/sys/param.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/newlib/libc/include/sys/param.h b/newlib/libc/include/sys/param.h
index 6460660..cf8a77c 100644
--- a/newlib/libc/include/sys/param.h
+++ b/newlib/libc/include/sys/param.h
@@ -5,11 +5,13 @@
 #ifndef _SYS_PARAM_H
 # define _SYS_PARAM_H
 
+#if !defined(__ASSEMBLY__)
 #include <sys/config.h>
 #include <sys/syslimits.h>
 #include <machine/endian.h>
 #include <machine/param.h>
 #include <limits.h>
+#endif
 
 #ifndef NBBY
 # define NBBY 8		/* number of bits in a byte */
-- 
2.43.0

