From 8ff22e9094549da550d55455bfd0e3958b9b3a45 Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Mon, 30 Jun 2025 04:58:05 +0200
Subject: [PATCH 20/25] Fix libc/stdio/rename.c

---
 newlib/libc/stdio/rename.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/newlib/libc/stdio/rename.c b/newlib/libc/stdio/rename.c
index 16ef3e3..d1eb9db 100644
--- a/newlib/libc/stdio/rename.c
+++ b/newlib/libc/stdio/rename.c
@@ -53,11 +53,11 @@ Supporting OS subroutines required: <<link>>, <<unlink>>, or <<rename>>.
 
 #ifndef _REENT_ONLY
 
-int
-rename (const char *old,
-       const char *new)
-{
-  return _rename_r (_REENT, old, new);
-}
+// int
+// rename (const char *old,
+//        const char *new)
+// {
+//   return _rename_r (_REENT, old, new);
+// }
 
 #endif
-- 
2.43.0

