From 4fabbb0a6561a556c4c26938f69436ca28cef371 Mon Sep 17 00:00:00 2001
From: Salar Samani <soltanisamani@gmail.com>
Date: Mon, 30 Jun 2025 05:06:06 +0200
Subject: [PATCH 23/25] libc/stdlib/abort.c

---
 newlib/libc/stdlib/abort.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/newlib/libc/stdlib/abort.c b/newlib/libc/stdlib/abort.c
index 065dee5..0ec001a 100644
--- a/newlib/libc/stdlib/abort.c
+++ b/newlib/libc/stdlib/abort.c
@@ -46,18 +46,18 @@ Supporting OS subroutines required: <<_exit>> and optionally, <<write>>.
 #include <unistd.h>
 #include <signal.h>
 
-void
-abort (void)
-{
-#ifdef ABORT_MESSAGE
-  write (2, "Abort called\n", sizeof ("Abort called\n")-1);
-#endif
-
-  while (1)
-    {
-      raise (SIGABRT);
-      _exit (1);
-    }
-}
+// void
+// abort (void)
+// {
+// #ifdef ABORT_MESSAGE
+//   write (2, "Abort called\n", sizeof ("Abort called\n")-1);
+// #endif
+
+//   while (1)
+//     {
+//       raise (SIGABRT);
+//       _exit (1);
+//     }
+// }
 
 #endif
-- 
2.43.0

